docs/threat-model.md

```diff
@@ -0,0 +1,297 @@
+# Threat model
+
+Initial threat model for `devops-deployments-metrics`.
+
+## Unauthorized Access
+
+### Threat
+
+Unauthorized users gaining access to the CLI application or sensitive information provided during runtime.
+
+### Mitigation
+
+Implement strong authentication and access controls to ensure only authorized users can execute the CLI application and
+access sensitive data. Consider utilizing secure token-based authentication or API keys.
+
+## Command Injection
+
+### Threat
+
+Attackers injecting malicious commands through user input, leading to unintended execution of arbitrary code.
+
+### Mitigation
+
+Apply proper input validation and sanitization techniques to prevent command injection attacks. Use parameterized queries
+or prepared statements when executing commands or interacting with external systems.
+
+## Insecure Credential Handling
+
+### Threat
+
+Improper handling of user credentials, such as GitHub username and password, leading to potential exposure or compromise.
+
+### Mitigation
+
+Implement secure credential management practices, such as securely storing credentials, avoiding hard-coding them within
+the source code, and utilizing secure credential storage mechanisms like environment variables or secret management solutions.
+
+### Threat
+
+Improper handling of user credentials, such as GitHub username and password, leading to potential exposure or compromise.
+
+### Mitigation
+
+Implement secure credential management practices, such as securely storing credentials, avoiding hard-coding them within
+the source code, and utilizing secure credential storage mechanisms like environment variables or secret management
+solutions. Ensure that sensitive credentials, such as the GitHub username and password, are properly protected and encrypted.
```

Comment:
You have two threats and two mitigations in this section that are awfully similar.

Reply:
I made some changes based on this feedback, Peter.